include/nrf_errno.h

```diff
@@ -0,0 +1,107 @@
+#define ERANGE 34           /**< Result too large */
+#define ENOMSG 35           /**< No message of the desired type */
+#define EDEADLK 45          /**< Resource deadlock would occur */
+#define ENOLCK 46           /**< No locks available */
+#define ENOSTR 60           /**< Not a STREAM */
+#define ENODATA 61          /**< No message is available on the STREAM head read queue */
+#define ETIME 62            /**< Stream ioctl() timeout */
+#define ENOSR 63            /**< No STREAM resources */
+#define EPROTO 71           /**< Protocol error */
+#define EBADMSG 77          /**< Bad message */
+#define ENOSYS 88           /**< Functionality not supported */
+#define ENOTEMPTY 90        /**< Directory not empty */
+#define ENAMETOOLONG 91     /**< File name too long */
+#define ELOOP 92            /**< Too many levels of symbolic links */
+#define EOPNOTSUPP 95       /**< Operation not supported on socket */
+#define ECONNRESET 104      /**< Connection reset */
+#define ENOBUFS 105         /**< No buffer space available */
+#define EAFNOSUPPORT 106    /**< Address family not supported */
+#define EPROTOTYPE 107      /**< Protocol wrong type for socket */
+#define ENOTSOCK 108        /**< Not a socket */
+#define ENOPROTOOPT 109     /**< Protocol not available */
+#define ECONNREFUSED 111    /**< Connection refused */
+#define EADDRINUSE 112      /**< Address in use */
+#define ECONNABORTED 113    /**< Connection aborted */
+#define ENETUNREACH 114     /**< Network unreachable */
+#define ENETDOWN 115        /**< Network is down */
+#define ETIMEDOUT 116       /**< Connection timed out */
+#define EHOSTUNREACH 118    /**< Host is unreachable */
+#define EINPROGRESS 119     /**< Operation in progress */
+#define EALREADY 120        /**< Connection already in progress */
+#define EDESTADDRREQ 121    /**< Destination address required */
+#define EMSGSIZE 122        /**< Message too large */
+#define EPROTONOSUPPORT 123 /**< Protocol not supported */
+#define EADDRNOTAVAIL 125   /**< Address not available */
+#define ENETRESET 126       /**< Connection aborted by network */
+#define EISCONN 127         /**< Socket is connected */
+#define ENOTCONN 128        /**< The socket is not connected */
+#define ENOTSUP 134         /**< Not supported */
+#define EILSEQ 138          /**< Illegal byte sequence */
+#define EOVERFLOW 139       /**< Value too large to be stored in data type */
+#define ECANCELED 140       /**< Operation canceled */
+
+#define EWOULDBLOCK EAGAIN /**< Operation would block */
+
+/* Not in IEEE Std 1003.1-2017 */
+
+#define ENOTBLK 15          /**< Block device required */
+#define EPFNOSUPPORT 96     /**< Protocol family not supported */
+#define ESHUTDOWN 110       /**< Can't send after socket shutdown */
+#define EHOSTDOWN 117       /**< Host is down */
+#define ESOCKTNOSUPPORT 124 /**< Socket type not supported */
+#define ETOOMANYREFS 129    /**< Too many references: can't splice */
```

Comment:
I would prefer to put them together in the same list as it is in Zephyr. The error codes must be the same as in Zephyr, so keeping the file similar to Zephyr makes it easier to keep them aligned.

Reply:
But it is just my opinion. I am not insisting on it.